test.js

```diff
@@ -2,95 +2,105 @@ var test = require('tape')
-    var key = v.key.length === 0 ? null : hexWrite(new Uint8Array(v.key.length / 2), v.key)
-    var salt = v.salt.length === 0 ? null : hexWrite(new Uint8Array(v.salt.length / 2), v.salt)
-    var personal = v.personal.length === 0 ? null : hexWrite(new Uint8Array(v.personal.length / 2), v.personal)
-
-    var expected = Buffer.from(hexWrite(new Uint8Array(v.out.length / 2), v.out))
-    var actual = Buffer.from(blake2b(out, input, key, salt, personal, true))
-
-    assert.deepEquals(actual, expected)
+setup()
+test('wait for ready', function (assert) {
+  blake2b.ready(function () {
+    assert.end()
   })
-  assert.end()
 })
+setup()
+
+function setup () {
+  test('vectors', function (assert) {
+    vectors.forEach(function (v) {
+      var out = new Uint8Array(v.outlen)
+      var input = hexWrite(new Uint8Array(v.input.length / 2), v.input)
+      var key = v.key.length === 0 ? null : hexWrite(new Uint8Array(v.key.length / 2), v.key)
+      var salt = v.salt.length === 0 ? null : hexWrite(new Uint8Array(v.salt.length / 2), v.salt)
+      var personal = v.personal.length === 0 ? null : hexWrite(new Uint8Array(v.personal.length / 2), v.personal)
+
+      var expected = Buffer.from(hexWrite(new Uint8Array(v.out.length / 2), v.out))
+      var actual = Buffer.from(blake2b(out.length, key, salt, personal, true).update(input).digest(out))
+
+      assert.deepEquals(actual, expected)
+    })
+    assert.end()
+  })
 
-test('works with buffers', function (assert) {
-  var vector = vectors.slice(-1)[0]
+  test('works with buffers', function (assert) {
+    var vector = vectors.slice(-1)[0]
 
-  var out = Buffer.allocUnsafe(vector.outlen)
-  var input = Buffer.from(vector.input, 'hex')
-  var key = Buffer.from(vector.key, 'hex')
-  var salt = Buffer.from(vector.salt, 'hex')
-  var personal = Buffer.from(vector.personal, 'hex')
+    var out = Buffer.allocUnsafe(vector.outlen)
+    var input = Buffer.from(vector.input, 'hex')
+    var key = Buffer.from(vector.key, 'hex')
+    var salt = Buffer.from(vector.salt, 'hex')
+    var personal = Buffer.from(vector.personal, 'hex')
 
-  var expected = Buffer.from(vector.out, 'hex')
-  var actual = blake2b(out, input, key, salt, personal)
+    var expected = Buffer.from(vector.out, 'hex')
+    var actual = blake2b(out.length, key, salt, personal).update(input).digest(out)
 
-  assert.deepEquals(actual, expected)
-  assert.end()
-})
+    assert.deepEquals(actual, expected)
+    assert.end()
+  })
 
-test('streaming', function (t) {
-  var isntance = blake2b.instance(blake2b.BYTES)
-  var buf = new Buffer('Hej, Verden')
+  test('streaming', function (t) {
+    var isntance = blake2b(blake2b.BYTES)
+    var buf = new Buffer('Hej, Verden')
 
-  for (var i = 0; i < 10; i++) isntance.update(buf)
+    for (var i = 0; i < 10; i++) isntance.update(buf)
 
-  var out = Buffer.alloc(blake2b.BYTES)
-  isntance.final(out)
+    var out = Buffer.alloc(blake2b.BYTES)
+    isntance.digest(out)
```

Comment:
i'm 99% sure i made this typo in another test. this clearly shows that @emilbayes stole that test. #watermarkhoneypot